### PR TITLE
Accordion Block enhancements

### LINKED ISSUE #24: AccordionBlock - enhancements
AccordionPanel - make versioned, add Link

### PR DESCRIPTION
**Pull request by jsirish**
AccordionBlock:
* JS requirements

AccordionPanel:
* added versioning
* added `BlockLink`

Template work

fixes #24
ping #4

### CODE REVIEW

File: code/blocks/AccordionBlock.php [hunk 1]
```diff
@@ -55,4 +55,20 @@ public function getCMSFields()
...
         return $fields;
     }
+}
+
+class AccordionBlock_Controller extends Block_Controller
+{
+    public function init()
+    {
+        Requirements::javascript(THIRDPARTY_DIR . '/jquery-ui/jquery-ui.js');
+        Requirements::customScript('
```
Comment (muskie9) [inline code/blocks/AccordionBlock.php]: Would it make sense to allow for customizing the css class for the accordion? Possibly a default config but a developer could update the default if they need to in the event of a class collision in their css.

Comment (jsirish) [reply code/blocks/AccordionBlock.php]: Yeah that makes sense. Cool if I add as a beta ticket?

Comment (muskie9) [reply code/blocks/AccordionBlock.php]: Ya, I think that's fine. Not a show stopper, just future flexibility.

**Review by muskie9 — APPROVED** [review]

**Comment by jsirish** [comment]
I was able to work in setting the accordion class via config. Can confirm it's overwriteable in project config

File: _config/config.yml [hunk 2]
```diff
@@ -12,4 +12,6 @@ SlideshowBlock:
     - FlexSlider
 SlideImage:
   extensions:
-    - DynamicBlocksSlideImageDataExtension
\ No newline at end of file
+    - DynamicBlocksSlideImageDataExtension
+AccordionBlock_Controller:
+  accordion_class: 'accordion_block'
```
Comment (muskie9) [inline _config/config.yml]: This should be deferred to the class declaration as it would be a duplicate.

File: code/blocks/AccordionBlock.php [hunk 3]
```diff
@@ -55,4 +55,48 @@ public function getCMSFields()
...
+        return $this->Panels();
+    }
+}
+
+class AccordionBlock_Controller extends Block_Controller
+{
+    /**
+     * @var string
+     */
+    private $accodion_class = 'accordion-block';
```
Comment (muskie9) [inline code/blocks/AccordionBlock.php]: I think we need the `static` declared as well for this to be used by the `AccordionClass()` method. This is also declared in the `yml` file, but that should be deferred to the php option I think: `private static $accordion_class = 'accordion-block';` here.

**Review by muskie9 — COMMENTED** [review]
I think the mix of the yml with the php allows for successful results, however, the `private` vs a `private static` makes the declaration in the controller class unused in the `AccordionClass()` method. I think the yml declaration can be deferred to a php declaration (just adding `static` to the `private` var) would work here.

**Pull request merged**